templatefitter/fit_model/model_builder.py

```diff
@@ -1098,8 +1115,6 @@ def _create_bin_nuisance_parameters(
                 parameter_type=ParameterHandler.bin_nuisance_parameter_type,
                 floating=True,
                 initial_value=initial_nuisance_value,
-                constrain_to_value=initial_nuisance_value,
```

Comment:
This might require some explanation and I'm not 100% sure about this but I think I'm correct: 
I think with the existing approach the bin nuisance parameters are basically constrained twice: Once via the `_gauss_term()` and once via the `_constraint_term()`, leading to an overall gaussian constraint *squared*. This can also be seen when producing pull plots for the nuisance parameters:

In the default behavior (here for the fit given in the examples but with 10x statistics in the templates) the pulls have uncertainties of 0.68 sigma:
![10x_weight_signal_pulls_double_constrained](https://user-images.githubusercontent.com/10749132/206688523-00f82a44-f703-4374-8de9-d211c7b62fdf.png)

With this change, they are 1 sigma as expected:
![10x_weight_signal_pulls_single_constrained](https://user-images.githubusercontent.com/10749132/206688785-a08787f6-2369-41b7-bec6-7dee838612c1.png)

I have pushed this change in this branch as this also should lead to some speedup as the constraint calculation can be skipped if no explicit constraints are given.

Again, as I said: I might be wrong here but I think this makes sense.

Reply:
Nice spot. I think this is a relic of another approach of the implementation of the nuisance parameters, but I am not sure.